src/03_output_crime_rate.R

```diff
@@ -0,0 +1,221 @@
+# install_github("bcgov/safepaths")
+pacman::p_load( tidyverse,config,bcmaps, bcdata, janitor,cansim,safepaths, arrow, duckdb,datadictionary)
+
+######################################################################################
+# Crime rate data
+# 
+######################################################################################
+
+# What’s cool is that Stats Can compiles on this data yearly at the police jurisdiction level (basically municipality level like Saanich): 
+# https://www150.statcan.gc.ca/t1/tbl1/en/tv.action?pid=3510018401. 
+
+########################################################################################################
+
+#  B.C. crime trends and statistics
+# https://www2.gov.bc.ca/gov/content/justice/criminal-justice/policing-in-bc/publications-statistics-legislation/crime-police-resource-statistics
+
+# Incident-based crime statistics, by detailed violations, police services in British Columbia 1, 2, 3, 4, 5
+# Frequency: Annual
+
+# Statistics Canada. Table 35-10-0184-01 Incident-based crime statistics, by detailed violations, police services in British Columbia, annual (number unless otherwise noted)
+
+
+# Release date: 2023-07-27
+
+
+
+# Geography: Province or territory, Policing district/zone
+
+# https://www2.gov.bc.ca/assets/gov/law-crime-and-justice/criminal-justice/police/publications/statistics/bc-crime-statistics-2022.xlsx
+
+# Policing district/zone is different from 
+
+# https://catalogue.data.gov.bc.ca/dataset/policing-jurisdictions-and-regions-in-bc
+
+
+# annual and policing district data. Luckily, BC stats team already aggregate the one data variable (total rate excluding traffic) to region level which is close to CD. 
+
+# https://www150.statcan.gc.ca/t1/tbl1/en/tv.action?pid=3510018401
+
+########################################################################################################
+print("file out of date: this could take a while")
+
+# Use Case
+# get_cansim_sqlite is particularly beneficial when working with large tables or when you need to perform complex filtering operations before bringing data into memory. For example, table 35-10-0184-01 is several gigabytes in size, making it impractical to load entirely into memory.
+# However, for smaller tables that can be easily processed in memory, get_cansim may still be a simpler and more straightforward option. The choice between the two functions depends on the specific requirements of your data analysis task and the size of the dataset you're working with.
+# it was supposed to save sqlite to a cache folder. 
+cansim_id <- "35-10-0184-01"
+# it is too slow to index in sqlite on a network drive, so switch to a local folder, and create a copy from LAN use_network_path("data/cansim_cache") to C drive repo_folder/data. 
+options(cansim.cache_path = "./data")
+# options(cansim.cache_path = use_network_path("data/cansim_cache"))
+getOption("cansim.cache_path")
+connection <- cansim::get_cansim_sqlite(cansim_id,
```

Comment:
I can't proceed!

```
Error in `.data$cansimTableNumber`:
! Column `cansimTableNumber` not found in `.data`.
Run `rlang::last_trace()` to see where the error occurred.
Warning messages:
1: In dir.create(cache_path) :
  cannot create dir '.\data\cansim_35100184_eng', reason 'No such file or directory'
2: In dir.create(base_dir) :
  cannot create dir '.\data\cansim_35100184_eng', reason 'No such file or directory'
```

Reply:
We need to copy the cached file from the LAN to the c: drive local repo folder. It is a huge file so it is impossible to work on it when it stays in the LAN directly.

Reply:
Ahhh...so that's a step I just have to do myself beforehand? Interesting.

Reply:
Would it make sense to add this line then?

```
fs::file_copy("G:/Operations/Data Science and Analytics/2024 SES Index/data/cansim_cache/35100184-eng.sqlite", "./data/")
```

Reply:
(But use the `safepaths` way)

Reply:
> Would it make sense to add this line then?
> 
> ```
> fs::file_copy("G:/Operations/Data Science and Analytics/2024 SES Index/data/cansim_cache/35100184-eng.sqlite", "./data/")
> ```

Wow. This is great idea with `safepaths` way.

Reply:
I have added it.